### ISSUE TITLE
[accessibility] public share: cannot use "download" button with keyboard only

### ISSUE THREAD

**Issue by susnux** [nextcloud/server#62245]
<img width="544" height="219" alt="Image" src="https://github.com/user-attachments/assets/e1148abc-851a-481d-83eb-0f253c48ac12" />

The download button can be focused using the keyboard but neither enter nor space triggers the download action.

**Comment by kristian-zendato** [comment]
@susnux I have reviewed this issue and would like to raise a few points for discussion.
The download button is implemented as an `<a>` tag which does not natively respond to the `Space` key event.
Could you advise whether the `Space` key event should also be enabled for this button?

**Comment by susnux** [comment]
>  Could you advise whether the Space key event should also be enabled for this button?

The element should be keyboard accessible just like any other link / button.
So in this case at least `enter` should work.

In theory there should be no need to do any special handling here the browser behavior should work, but something prevents the default behavior
